crates/runc-shim/src/main.rs

```diff
@@ -27,7 +29,21 @@ mod task;
 
 use service::Service;
 
+fn parse_version() {
+    let os_args: Vec<_> = env::args_os().collect();
+    let flags = parse(&os_args[1..]).unwrap();
+    if flags.version {
+        println!("{}:", os_args[0].to_string_lossy());
+        println!("  Version: {}", env!("CARGO_PKG_VERSION"));
+        println!("  Revision: {}", env!("CARGO_GIT_HASH"));
+        println!();
+
+        std::process::exit(0);
+    }
+}
+
 #[tokio::main]
 async fn main() {
+    parse_version();
```

Comment:
Why not put it to `bootstrap`, if not, flags will be parsed twice.
And this version flag should be a common code in shim, not in runc-shim

Reply:
I still prefer to put it to `bootstrap` for the clean code principle. This code can be uesed by other shim too. We can add another custom env to store the runc-shim pkg version and then print it in shim.

Reply:
>  We can add another custom env to store the runc-shim pkg version and then print it in shim.

Could you please elaborate on this point? I just can't figure out how the env variables built in `runc-shim` could be used in `shim` crate. `runc-shim` is depending on `shim`, so rustc will build `shim` first, and then `runc-shim` second. When building `shim` crate, rustc will not be able to see the env variables that `runc-shim` adds. Am I missing something here?

Reply:
I got you. You are right. If we only provide git hash id, we can move all the parse_version in `shim`, but with pkg version, that's pretty hard!